crates/chia-sdk-driver/src/layers/datalayer/oracle_layer.rs

```diff
@@ -0,0 +1,94 @@
+use chia_protocol::{Bytes, Bytes32};
+use chia_sdk_types::{Condition, CreateCoin, CreatePuzzleAnnouncement};
+use clvm_traits::{clvm_quote, match_quote, FromClvm, ToClvm};
+use clvmr::{Allocator, NodePtr};
+
+use crate::{DriverError, Layer, Puzzle, Spend, SpendContext};
+
+/// The Oracle [`Layer`] enables anyone to spend a coin provided they pay an XCH fee to an address.
+/// It's typically used with [`DelegationLayer`](crate::DelegationLayer).
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct OracleLayer {
+    /// The puzzle hash corresponding to the address the fee should be paid to.
+    pub oracle_puzzle_hash: Bytes32,
+    /// The amount of XCH that should be paid to the oracle.
+    pub oracle_fee: u64,
+}
+
+impl OracleLayer {
+    #[allow(clippy::missing_panics_doc)]
+    pub fn new(oracle_puzzle_hash: Bytes32, oracle_fee: u64) -> Self {
+        assert!(oracle_fee % 2 == 0, "oracle fee must be even");
```

Comment:
Instead of this panic, maybe an error would make sense here or returning `Option` and adding a doc comment explaining this must be even.